stripe-ui-core/src/main/java/com/stripe/android/uicore/image/StripeImageLoader.kt

```diff
@@ -0,0 +1,164 @@
+package com.stripe.android.uicore.image
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import androidx.annotation.RestrictTo
+import androidx.annotation.WorkerThread
+import com.stripe.android.core.Logger
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.withContext
+import java.net.URL
+import java.util.concurrent.ConcurrentHashMap
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+/**
+ * Image loader that fetches images from memory, disk or network and runs
+ * cache policy accordingly.
+ *
+ * [StripeImageLoader] is stateful as it holds the memoryCache instance. For
+ * memory cache to work the image loader instance needs to be shared.
+ *
+ * @param memoryCache, memory cache to be used, or null if no memory cache is desired.
+ * @param diskCache, memory cache to be used, or null if no memory cache is desired.
+ */
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class StripeImageLoader(
```

Comment:
Can you add some tests?

Reply:
https://github.com/stripe/stripe-android/pull/5307/commits/6056ea389bf826842eeebd13eb87568e21746326